### ISSUE TITLE
Unable to run libinput-gestures with Plasma 5

### ISSUE THREAD

**Issue by adrfantini** [bulletmark/libinput-gestures#4]
Hi and thanks for this neat tool!
I have one small problem: I can run the program manually from the terminal, but for some reason when I try to launch it from the DE (plasma 5) I get this output: (see attached screenshot)
![screenshot_20160307_170004](https://cloud.githubusercontent.com/assets/12826141/13575140/869211d2-e487-11e5-9e63-64782272942f.png)
It's a really small niglle, I just thought I could let you know. Thanks again.

**Comment by bulletmark** [comment]
I don't use KDE but this looks like some odd KDE requirement. I see essentially the same problem raised here http://askubuntu.com/questions/738114/packet-tracer-6-x-on-kubuntu-15-10-unfind-libkdeinit5-packettracer6-so and here https://www.kubuntuforums.net/showthread.php?67957-Cannot-Launch-System-Settings although it is hard to see how the solutions there apply here. The desktop file may be missing something KDE requires? I don't want to install KDE to debug this so perhaps you can try?

**Comment by adrfantini** [comment]
I'd be glad to help but I really don't know where to start from.

2016-03-08 2:46 GMT+01:00 Mark Blakeney <notifications@github.com>:

**Comment by rickysarraf** [comment]
@adrfantini  Can you share the output of the following ?


```
ldd `which libinput-gestures`
```

My guess is KDE is shipping something with the same name.

**Comment by adrfantini** [comment]
Sure thing. Maybe I did something wrong during installation, but
libinput-gestures is not in my PATH:

which libinput-gestures                                         [1]
which: no libinput-gestures in
(/usr/local/sbin:/usr/local/bin:/usr/bin:/usr/lib/j
vm/default/bin:/usr/bin/site_perl:/usr/bin/vendor_perl:/usr/bin/core_perl:/usr/loc
al/bin)

Thus ldd returns only errors.


2016-03-08 14:16 GMT+01:00 Ritesh Raj Sarraf <notifications@github.com>:

**Comment by rickysarraf** [comment]
But you said that you were able to run the program from the terminal??

s3nt fr0m a $martph0ne, excuse typ0s

**Comment by adrfantini** [comment]
Well, "manually" from the terminal, that is running the executable
directly. Sorry, that was not clear at all.

2016-03-08 16:29 GMT+01:00 Ritesh Raj Sarraf <notifications@github.com>:

> But you said that you were able to run the program from the terminal??
>
> s3nt fr0m a $martph0ne, excuse typ0s

**Comment by bulletmark** [comment]
That may be your problem then. `make install` installs _libinput-gestures_ to your `~/bin` dir which I assumed was always in a user's path but it does not look to be by default. So add the following to your `~/.profile` or `~/.bash_profile` then log out and back in again. Make sure you do a `make install` from the `libinput-gestures` dir of course.
~~~~
if [ -d "$HOME/bin" ] ; then
    export PATH="$HOME/bin:$PATH"
fi
~~~~

**Comment by adrfantini** [comment]
This AFAIK is not a general behaviour, it's just one that Ubuntu and other
distros enforce. I just tried two Mint machines and it was not set (old
versions of Mint tho).
You may want to add that to the installation instructions or ask the user
to move the executable to a directory within the $PATH.

2016-03-09 2:11 GMT+01:00 Mark Blakeney <notifications@github.com>:

**Comment by rickysarraf** [comment]
Hello,

On Wed, 2016-03-09 at 04:37 -0800, adrfantini wrote:
> This AFAIK is not a general behaviour, it's just one that Ubuntu and
> other
> distros enforce. I just tried two Mint machines and it was not set
> (old
> versions of Mint tho).

Maybe newer versions of Bash may have rolled it. On my Debian box
/etc/skel/.profile sets user home ~/bin into PATH. And that file is
reported to be owned by bash


rrs@learner:/etc$ dpkg -S /etc/skel/.profile 
bash: /etc/skel/.profile
2016-03-09 / 18:56:40 ♒♒♒  ☺  


> You may want to add that to the installation instructions or ask the
> user
> to move the executable to a directory within the $PATH.

I think this assumption is much simpler. adding a binary to a system
path would mean root/sudo privileges.

Maybe just documenting both should suffice.


But, back to your problem, does setting the path solve it ?
I'm still curious about how KDE generated that error message. :-)

-- 
Ritesh Raj Sarraf
RESEARCHUT - http://www.researchut.com
"Necessity is the mother of invention."

**Comment by adrfantini** [comment]
Setting ~/bin in the $PATH indeed solves the problem.

2016-03-09 14:30 GMT+01:00 Ritesh Raj Sarraf <notifications@github.com>:

**Comment by bulletmark** [comment]
Fixed with commit 1c7a6c6f8183b85bfa9b37af3e0d56d4b7757299.

The problem was the desktop file was relying on the user having his PATH include `~/bin` but I have changed the installer to insert the full path to the executable in the desktop file.
